### PR TITLE
tests: Add core protocol tests (manual, Screens section)

### PR DESCRIPTION
**Pull request by uint**
Mostly basic type checking for screen-related functions. I actually didn't manage to find any problems. At the very least, all the specified functions seem to work exactly as described in the reference.

### CODE REVIEW

File: tests/core-protocol.lisp [hunk 1]
```diff
@@ -66,3 +66,42 @@
     (is (null (xlib:close-display display)))
     ;; We can't query closed display.
     (signals xlib:closed-display (xlib:query-extension display "FOO"))))
```
Comment (dkochmanski) [inline tests/core-protocol.lisp]: change test-package `:documentation` option (that it tests core protocol, not section 3 only)

Comment (dkochmanski) [reply tests/core-protocol.lisp]: I mean at the top of the file.
Also:
- are these all functions from chapter 4?
- did you find any mistakes in the manual?

Comment (uint) [reply tests/core-protocol.lisp]: These are all the functions from chapter 4. I found no mistakes, although I have been looking for them. I'll push a squashed commit once I run the tests on ECL and CCL to make sure all's good.

**Comment by uint** [comment]
Okay, so the tests pass on SBCL, CCL, ECL.

```
tom@tomian:~$ sbcl --version
SBCL 1.4.2.debian
tom@tomian:~$ ecl --version
ECL 16.1.3
tom@tomian:~$ /opt/ccl/lx86cl64 --version
Version 1.12-dev-r16752  (LinuxX8664)
```

I've also made the changes you requested and squashed the commits.

**Comment by dkochmanski** [comment]
Thanks! merging.

**Pull request merged**